### DISCUSSION TITLE
It should be possible to read from a SQLite database into a DuckDB table if the latter is correctly defined

### DISCUSSION THREAD

**Discussion by pkoppstein** [duckdb/duckdb#13539] [Ideas]
The DuckDB documentation states:

"When querying SQLite, DuckDB must deduce a specific column type mapping."

Unfortunately, the rules for "deducing" the type of a column are
currently being applied even when the DuckDB table makes the type
explicit, i.e. when "deduction" is not needed.  This hinders reading SQLite databases 
and leads to paradoxical situations where for example the number of rows in a table can be correctly
determined by DuckDB but other queries fail.

Example:

Consider the "well-known SQLITE database" at https://www.kaggle.com/datasets/hugomathien/soccer
After downloading and unzipping the file as database.sqlite:

ATTACH 'database.sqlite' AS db (TYPE sqlite); USE db; select count() from Player;

produces the correct count:
```
┌──────────────┐
│ count_star() │
│    int64     │
├──────────────┤
│        11060 │
└──────────────┘
```

As indicated above, this in itself is puzzling 
because the SQLite schema for the Players table declares
`height` to be INTEGER whereas it in fact contains floats.

That is, in accordance with the DuckDB documentation regarding
the way DuckDB infers types, one might expect an error to have been raised
as soon as the Player table was read.  

In any case, when the `height` column is read, an error is detected:

```
select distinct(height) from Player;
```
produces
```
Mismatch Type Error: Invalid type in column "height": column was declared as integer, found "182.88" of type "float" instead.
```

Now one might hope that declaring the type of the field to be FLOAT
before importing it would circumvent the need for type inference, but apparently that is not the case:
```
LOAD sqlite;

CREATE OR REPLACE TABLE Player (
    id BIGINT,
    player_api_id BIGINT,
    player_name VARCHAR,
    player_fifa_api_id BIGINT,
    birthday VARCHAR,
    height FLOAT,
    weight BIGINT
);

INSERT INTO Player
FROM sqlite_scan('database.sqlite', 'Player');

· Mismatch Type Error: Invalid type in column "height": column was declared as integer, found "182.88" of type "float" instead.
```
### Versions tried:
v1.0.0   
v1.0.1-dev4509

### Circumventing the issue

Fortunately, one can circumvent the problem by first running:
```
SET GLOBAL sqlite_all_varchar = true;
```

**Comment by chapmanjacobd** [comment]
I agree, if the type is set already then there should, at the very least, be an option to SAFE_CAST() all the data into the existing type.

After importing with `sqlite_all_varchar` you can try using this to identify data type issues: 

https://github.com/chapmanjacobd/computer/blob/main/bin/duckdb_typecaster.py
